src/frontend/js/utils/test/wrappers/WrapInJoanieApp.tsx

```diff
@@ -0,0 +1,37 @@
+import { CunninghamProvider } from '@openfun/cunningham-react';
+import { PropsWithChildren } from 'react';
+import fetchMock from 'fetch-mock';
+import JoanieSessionProvider from 'contexts/SessionContext/JoanieSessionProvider';
+import { WrapInIntl } from './WrapInIntl';
+import { WrapInReactQuery } from './WrapInReactQuery';
+import { WrapInRouter } from './WrapInRouter';
+import { WrapInAppProps } from './types';
+
+export const nbSessionApiRequest = 3;
+beforeEach(() => {
+  // JoanieSessionProvider inital requests
+  fetchMock.get('https://joanie.endpoint/api/v1.0/orders/', []);
+  fetchMock.get('https://joanie.endpoint/api/v1.0/addresses/', []);
+  fetchMock.get('https://joanie.endpoint/api/v1.0/credit-cards/', []);
```

Comment:
I think there will be conflicts when we would like to override those routes.

Reply:
yes, we can use the override option. it's a trade of.

Reply:
I moved it in a `setupJoanieSession` function that now need to be called on each test.
It loose some magic but make it easier to track and keep the session's mock creation in a single place.